l10n_it_fatturapa_out/wizard/efattura.py

```diff
@@ -255,9 +255,27 @@ def get_importo(line):
                 return False
             return line.price_unit * line.discount / 100
 
-        def get_default_note_tax(record):
-            all_taxes = get_all_taxes(record)
-            return all_taxes[0]["AliquotaIVA"] if all_taxes else "0.00"
+        def get_tax_ids(line):
+            """Get the tax for an invoice line.
+
+            Note: SdI expects only one tax per line. Also, it expects to find a tax line
+            for each line mentioned in the invoice lines, but Odoo generate no tax line
+            if the amount is 0. This is handled in get_all_taxes().
+
+            Common best practice for note/section lines is to reference an existing tax
+            line, instead of creating a new zero tax summary in the XML. The price for
+            note/section lines is zero anyway.  Therefore the original tax is ignored as
+            it defaults to the company default tax which may or may not be non zero in
+            the invoice."""
+
+            if line.display_type in ("line_section", "line_note"):
+                # find a product line, with taxes, if possible
+                tax_lines = line.move_id.line_ids.filtered(
+                    lambda line: line.tax_ids and line.product_id
```

Comment:
Come detto in chiamata stamattina, manca da fare un'implementazione più direttamente collegata al controllo che fa lo SDI: andrebbe presa un'imposta qualsiasi dal riepilogo che generi in `get_all_taxes`.

Reply:
Fatto.